File: backend/app/schemas/system.py
from pydantic import BaseModel
from typing import Optional, Literal

class SystemSettings(BaseModel):
    disable_sql_logs: Optional[bool] = None
    log_level: Optional[Literal["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]] = None

class SystemSettingsResponse(BaseModel):
    settings: SystemSettings
    message: str

File: frontend/services/system.ts
import { get, put } from './api';
import { ApiResponse } from '@/types';

export type LogLevel = 'DEBUG' | 'INFO' | 'WARNING' | 'ERROR' | 'CRITICAL';

export interface SystemSettings {
  disable_sql_logs?: boolean;
  log_level?: LogLevel;
}

export interface SystemSettingsResponse {
  settings: SystemSettings;
  message: string;
}

/**
 * Get current system settings
 */
export const getSystemSettings = async (): Promise<ApiResponse<SystemSettingsResponse>> => {
  return await get<SystemSettingsResponse>('/system');
};

/**
 * Update system settings
 */
export const updateSystemSettings = async (
  settings: SystemSettings
): Promise<ApiResponse<SystemSettingsResponse>> => {
  return await put<SystemSettingsResponse>('/system', settings);
};